### PR TITLE
MNT Added PR template

### PR DESCRIPTION
**Pull request by merveenoyan**
Partially solves #191. I used a small part of @gradio-app/gradio's template and made changes. @skops-dev/maintainers feel free to add other things on top.
(Also I think this is all you need to do if you want a template, I followed this tutorial https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository let me know if there's anything I need to do on top) We can do the same for issues as well.

### CODE REVIEW

**Review by adrinjalali — COMMENTED** [review]
I kinda prefer this as a template: 

https://raw.githubusercontent.com/scikit-learn/scikit-learn/main/.github/PULL_REQUEST_TEMPLATE.md

**Comment by merveenoyan** [comment]
@adrinjalali I added more stuff. Feel free to tell me to add anything else if you want it.

File: .github/workflows/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+- [ ] I have performed a self-review of my own code
+- [ ] I have run style formatting on my code
+- [ ] I have commented my code in hard-to-understand areas
+- [ ] I have made changes to docstrings and notebooks
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] I added my changes to changelog in docs/changes.rst.
+
+Don't hesitate to ping @skops-dev/maintainers in your issues and pull requests if you don't receive a review in a timely manner. We try to review all pull requests as soon as we can.
```
Comment (adrinjalali) [inline .github/workflows/PULL_REQUEST_TEMPLATE.md:27]: I don't think this is useful. It sounds a bit condescending to me.

File: .github/workflows/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Thanks a lot for contributing to skops! Please make sure you've taken a look at the contribution guidelines: https://github.com/skops-dev/skops/blob/main/CONTRIBUTING.rst You can find answers to your questions about the checklist below in this file.
+Below is a template you can use to create pull requests.
```
Comment (adrinjalali) [inline .github/workflows/PULL_REQUEST_TEMPLATE.md:2]: line too long

**Review by adrinjalali — COMMENTED** [review]
all this text should also be hidden in the comment tag (as is in the link on the sklearn side).

I would start with the sklearn template and change that instead of using the gradio one. I think that one is a better written one.

File: .github/workflows/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (adrinjalali) [reply .github/workflows/PULL_REQUEST_TEMPLATE.md:27]: "I performed a self review of my own code"? That's really bad. Also, not ever PR needs a change log. Also, we send PRs w/o the tests passing all the time, and that's fine, we fix them on the PR.

File: .github/workflows/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (merveenoyan) [reply .github/workflows/PULL_REQUEST_TEMPLATE.md:2]: Why do we have to apply line trimming on markdown doc that will appear on github inside PR opening page? I don't get it.

File: .github/workflows/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<!--
+Thanks for contributing a pull request! Please ensure you have taken a look at
+the contribution guidelines: https://github.com/skops-dev/skops/blob/main/CONTRIBUTING.rst
+-->
```
Comment (adrinjalali) [inline .github/workflows/PULL_REQUEST_TEMPLATE.md:4]: you could add an explicit note here about linting (black, isort, pre-commit hooks, etc) maybe?

**Review by adrinjalali — APPROVED** [review]
Thanks @merveenoyan . Let's merge and see if it helps.

**Pull request merged**